### PR TITLE
Architecture: Authzone as RENode

### PR DESCRIPTION
**Pull request by talekhinezh**

### CODE REVIEW

File: radix-engine/src/engine/modules/auth_module.rs [hunk 1]
```diff
@@ -19,16 +19,47 @@ impl AuthModule {
...
+            .find(|e| {
+                matches!(
+                    e,
+                    HeapRootRENode {
+                        root: HeapRENode::AuthZone(..),
+                        ..
+                    }
+                )
+            })
+            .expect("Could not find auth zone")
```
Comment (iamyulong) [inline radix-engine/src/engine/modules/auth_module.rs:38]: I think we should create an `AuthZone` even if the receiver is `&AuthZone` to have a simplified invariant - every frame has a call frame. :)

File: radix-engine/src/engine/node.rs [hunk 2]
```diff
@@ -291,6 +293,20 @@ impl HeapRENode {
...
+        match self {
+            HeapRENode::AuthZone(auth_zone, ..) => auth_zone,
+            _ => panic!("Expected to be a resource manager"),
+        }
+    }
+
+    pub fn auth_zone_mut(&mut self) -> &mut AuthZone {
+        match self {
+            HeapRENode::AuthZone(auth_zone, ..) => auth_zone,
+            _ => panic!("Expected to be a resource manager"),
```
Comment (iamyulong) [inline radix-engine/src/engine/node.rs:306]: Needs to update panic messages.

File: radix-engine/src/fee/fee_table.rs [hunk 3]
```diff
@@ -49,11 +70,15 @@ pub enum SystemApiCostingEntry<'a> {
...
+    ReadBlob {
+        size: u32,
+    },
     /// Generates a UUID.
     GenerateUuid,
     /// Emits a log.
-    EmitLog { size: u32 },
+    EmitLog {
+        size: u32,
+    },
```
Comment (iamyulong) [inline radix-engine/src/fee/fee_table.rs:81]: Minor: I guess we have a different formatter :)

**Review by iamyulong — APPROVED** [review]
Looks better than what's presented during our last discussion. 👍

File: radix-engine/src/engine/modules/auth_module.rs (hunk 1 above)
Comment (talekhinezh) [reply radix-engine/src/engine/modules/auth_module.rs:38]: sure :)

**Pull request merged**